pkg/transaction/transaction.go

```diff
@@ -291,11 +291,10 @@ func (t *transactionService) prepareTransaction(ctx context.Context, request *Tx
 		if err != nil {
 			return nil, err
 		}
-	}
-	gasPrice = new(big.Int).Div(new(big.Int).Mul(big.NewInt(int64(tipPercent)+100), gasPrice), big.NewInt(100))
-	if gasPrice.Cmp(minGasPrice) < 0 {
+	} else if gasPrice.Cmp(minGasPrice) < 0 {
 		return nil, ErrGasPriceTooLow
 	}
+	gasPrice = new(big.Int).Div(new(big.Int).Mul(big.NewInt(int64(tipPercent)+100), gasPrice), big.NewInt(100))
```

Comment:
Wait, you're going to boost the provided price?   Shouldn't the boost be inside the gasPrice == nil block?  If the user specifies a price, then I don't think they'd expect it to be boosted before being used.